pytorch_lightning/trainer/training_io.py

```diff
@@ -370,7 +370,7 @@ def dump_checkpoint(self, weights_only: bool = False) -> dict:
             if hasattr(model, '_hparams_name'):
                 checkpoint[LightningModule.CHECKPOINT_HYPER_PARAMS_NAME] = model._hparams_name
             # add arguments to the checkpoint
-            checkpoint[LightningModule.CHECKPOINT_HYPER_PARAMS_KEY] = model.hparams
+            checkpoint[LightningModule.CHECKPOINT_HYPER_PARAMS_KEY] = dict(model.hparams)
```

Comment:
I'm afraid this will break the omegaconf logic below...

Reply:
I'm not familiar with omegaconf, does it require the attribute style access that pl AttributeDict provides?

Should I just move this change to an else block after `if OMEGACONF_AVAILABLE`?

Reply:
well, go way would be having a test to check that it does not break...

Reply:
I'll go through existing tests and see if omegaconf hparams are covered when I get a chance

Reply:
There is already a `test_omega_conf_hparams` which checks it. Without the latest commit tests passed as well. I just checked the `load_from_checkpoint` function and it casts hparams to the correct type specified in the checkpoint (in the case of my initial implementation: dict -> Container). With the latest commit omegaconfs will (continue to) be pickled as Container.

I haven't used omegaconfs before so perhaps there might be something I'm missing.